lib/gitsh/cli.rb

```diff
@@ -82,5 +84,15 @@ def option_parser
         end
       end
     end
+
+    def ensure_executable_git
+      IO.popen(env.git_command).close
+    rescue Errno::ENOENT
+      env.puts_error(
+        "gitsh: No such executable git: #{env.git_command}. Please " \
+        'specify an executable git using the --git option.',
```

Comment:
Since this is specifically for `ENOENT`, I wonder if we should drop the mention of “executable”? How about:

    gitsh: #{env.git_command}: No such file or directory
    Ensure git is on your PATH, or specify the path to git using the --git option

The first line of this would be exactly what `ls`, `cat`, etc. would output, which seems like a nice consistency to aim for.